resources/config/varnish/fos_tags_xkey.vcl

```diff
@@ -0,0 +1,30 @@
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import xkey;
+
+sub fos_tags_xkey_recv {
+    if (req.method == "PURGEKEYS") {
+        if (!client.ip ~ invalidators) {
+            return (synth(405, "Not allowed"));
+        }
+
+        // If tag invalidation api allows caller to specify expire/purge in the future,
+        // then adjust this to be able to handle both headers in same request.
```

Comment:
@dbu NOTE: I was thinking of proposing to do a `ExpireTagCapable` for instance to have a method that clearly means expire and not purge, but then would need to create new `VarnishXkey` proxy client and so on as Ban would not support that. So just left this here for a possible 3.0 cleanup where things are aligned across newer proxies.

Reply:
i think its good to have both handled in this same vcl snippet. are both purge and softpurge available with all versions? as in: does this vcl always compile when the xkey extension is available?

is there a use case to mix expire and purge? if so, we could change the code below to use just `if` instead of `elseif` to handle both headers.

Reply:
> are both purge and softpurge available with all versions? as in: does this vcl always compile when the xkey extension is available?

yes and yes

> is there a use case to mix expire and purge?

right now it  is not possible, both in terms of design of tag invalidation api and as long as FOS emits queued url request for each call to invalidate.

So I'll remove the comment, it was anyway more for your eyes to be aware of this being possible if we allow for it.